### PR TITLE
Add packaging with poetry

### PR DESCRIPTION
**Pull request by sandrobonazzola**
Add packaging using poetry

Help installing the package as `pip install git+https://github.com/distributed-system-analysis/smallfile.git@main`
Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

### CODE REVIEW

**Review by portante — CHANGES_REQUESTED** [review]
Let's also drop `setup.cfg` and `requirements.txt` (making sure they are both properly covered by `pyproject.toml`), and then update the `README.md` where necessary to describe the use of `poetry`.

**Comment by sandrobonazzola** [comment]
> Let's also drop `setup.cfg` and `requirements.txt` (making sure they are both properly covered by `pyproject.toml`), and then update the `README.md` where necessary to describe the use of `poetry`.

Ok for dropping `requirements.txt` but we can't remove `setup.cfg`because pycodestyle and flake8 doesn't accept configuration via `pyproject.toml` yet.

File: setup.cfg [hunk 1]
```diff
@@ -1,7 +1,9 @@
 [pycodestyle]
+# pycodestyle doesn't support pyproject.toml yet
```
Comment (portante) [inline setup.cfg:2]: Where is `pycodestyle` invoked?  I only see the configuration here.  The pre-commit hook has `black` in it and `isort`.  We should probably add a GitHub action to run `black` and `isort` checks, and add a flake8 check to both as well.

File: setup.cfg [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
 max-line-length = 160
 
 [flake8]
+# flake8 doesn't support pyproject.toml yet
```
Comment (portante) [inline setup.cfg:6]: I believe `flake8` can use a `.flake8` file instead so that we can make it specific to `flake8`.

**Review by portante — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 1 above)
Comment (sandrobonazzola) [reply setup.cfg:2]: `pycodestyle` is taken in by `flake8` run. I think we can override the pycodestyle config with the flake8 one.

Comment (sandrobonazzola) [reply setup.cfg:2]: The GitHub action already exists: https://github.com/distributed-system-analysis/smallfile/blob/main/.github/workflows/code_check.yaml

**Review by portante — APPROVED** [review]

**Pull request merged**